spec/controllers/users_controller_spec.rb

```diff
@@ -27,6 +27,32 @@
       expect(response).to have_http_status(:bad_request)
       expect(user.reload.name).to eq(user.name)
     end
+
+    it 'updates the avatar' do
+      user = create(:user)
+      patch :update, params: { id: user.id, user: { avatar: fixture_file_upload(file_fixture('default-avatar.png'), 'image/png') } }
+      expect(user.reload.avatar).to be_attached
+    end
+
+    it 'deletes the avatar' do
+      user = create(:user)
+      user.avatar.attach(io: fixture_file_upload('default-avatar.png'), filename: 'default-avatar.png', content_type: 'image/png')
+      expect(user.reload.avatar).to be_attached
+      delete :purge_avatar, params: { id: user.id }
+      expect(user.reload.avatar).not_to be_attached
+    end
+
+    it 'returns an error if the avatar is a pdf' do
+      user = create(:user)
+      patch :update, params: { id: user.id, user: { avatar: fixture_file_upload(file_fixture('default-pdf.pdf'), 'pdf') } }
+      expect(user.reload.avatar).not_to be_attached
```

Comment:
Not sure if important, but if there is a way to check for an error occurring instead of the avatar not being attached it would be more specific.

Or just change ```'returns an error if the avatar is a pdf'``` to ```doesn't attach avatar is pdf``` or something similiar

Reply:
Yeah, you're right.